fix(lua): add keyword, number matching

davidhcefx opened the pull request:
Fix issue https://github.com/scopatz/nanorc/issues/393

- `goto` keyword
- `math.sin` typo
- Properly match numbers according to https://www.lua.org/manual/5.4/manual.html#3.1

davidhcefx committed:
fix(lua): add keyword, number matching

- goto keyword
- math.sin typo
- Properly match numbers according to https://www.lua.org/manual/5.4/manual.html#3.1

galenguyer commented on lua.nanorc:
```diff
@@ -65,7 +67,7 @@ color red "\"(\\.|[^\\\"])*\"|'(\\.|[^\\'])*'"
 
 # Escapes
-color red "\\[0-7][0-7][0-7]|\\x[0-9a-fA-F][0-9a-fA-F]|\\[abefnrs]|(\\c|\\C-|\\M-|\\M-\\C-)."
+#color red "\\[0-7][0-7][0-7]|\\x[0-9a-fA-F][0-9a-fA-F]|\\[abefnrs]|(\\c|\\C-|\\M-|\\M-\\C-)."
```
Any reason you've removed the escapes highlighting?

davidhcefx replied:
Opps, I migrated from the original author of this fix without looking into this line. It should be remained plus the character set `[abefnrs]` should also be updated. Thanks for the review!

davidhcefx committed:
fix(lua): update escape and adjust coloring

- Numbers and escapes: magenta

davidhcefx commented:
* Updated the rules, for example to include `\t` escape.
* Changed escape highlighting from `red` to `magenta`, because otherwise the highlighted result would be invisible.

### Before
<img src="https://github.com/galenguyer/nano-syntax-highlighting/assets/23246033/46661956-308b-4832-b2dc-96e98f204cd4" width=50%>

### After
<img src="https://github.com/galenguyer/nano-syntax-highlighting/assets/23246033/40fa37c8-c8d4-41a7-9355-1a6b27069df7" width=50%>
